bee-network/bee-gossip/src/peer/list.rs

```diff
@@ -112,18 +105,36 @@ impl PeerList {
         self.peers
             .get(peer_id)
             .ok_or(Error::PeerNotPresent(*peer_id))
-            .map(|(peer_info, _)| peer_info.clone())
+            .map(|(info, _, _)| info.clone())
+    }
+
+    pub fn metrics(&self, peer_id: &PeerId) -> Result<PeerMetrics, Error> {
+        self.peers
+            .get(peer_id)
+            .ok_or(Error::PeerNotPresent(*peer_id))
+            .map(|(_, _, metrics)| metrics.clone())
     }
 
     pub fn len(&self) -> usize {
         self.peers.len()
     }
 
+    /// Note: Returns an error if the address trying to be added is a duplicate.
+    pub fn add_local_addr(&mut self, addr: Multiaddr) -> Result<(), (Multiaddr, Error)> {
+        if self.local_addrs.contains(&addr) {
+            return Err((addr.clone(), Error::AddressIsDuplicate(addr)));
+        }
+
+        let _ = self.local_addrs.insert(addr);
```

Comment:
Why not use the result to conditionally return ? One less hash.

Reply:
you're right...I guess it used to be a HashMap before at some point, and that would have replaced the value.